src/NumericalAlgorithms/RootFinding/NewtonRaphson.hpp

```diff
@@ -17,102 +18,149 @@
 namespace RootFinder {
 /*!
  * \ingroup NumericalAlgorithmsGroup
- * \brief Finds the root of the function `f` with the Newton-Raphson method.
+ * \brief Finds the root of the function `f` with the Newton-Raphson
+ * method, falling back to bisection on poor convergence.
  *
- * `f` is a unary invokable that takes a `double` which is the current value at
- * which to evaluate `f`. `f` must return a `std::pair<double, double>` where
- * the first element is the function value and the second element is the
- * derivative of the function.  An example is below.
+ * `f` is a unary invokable that takes a `double` which is the current
+ * value at which to evaluate `f`. `f` must return a
+ * `std::pair<double, double>` where the first element is the function
+ * value and the second element is the derivative of the function.
+ * The method converges when the residual is smaller than or equal to
+ * \p residual_tolerance or when the step size is smaller than \p
+ * step_absolute_tolerance or the proposed result times \p
+ * step_relative_tolerance.
  *
  * \snippet Test_NewtonRaphson.cpp double_newton_raphson_root_find
  *
```

Comment:
Do you think it would be useful to add a note to the dox as to why we don't just use the Boost implementation? I'm fine either way :)

Reply:
Yes, I like this suggestion